File: src/SkyBlock/enchants/armor/Poisoned.php
<?php


namespace SkyBlock\enchants\armor;


use pocketmine\entity\effect\EffectInstance;
use pocketmine\entity\effect\VanillaEffects;
use pocketmine\player\Player;

class Poisoned extends BaseArmorEnchant {

    static int $id = 125;

    /**
     * @param int $level
     *
     * @return int
     */
    public function getLevel(int $level = 1) : int {
        return (($int = ceil($level / 3)) > 5) ? 5 : $int;
    }

    /**
     * @param Player $attacker
     * @param Player $victim
     * @param int    $enchantmentLevel
     */
    public function onActivation(Player $attacker, Player $victim, int $enchantmentLevel) : void {
        $attacker->getEffects()->remove(VanillaEffects::REGENERATION());
        $attacker->getEffects()->remove(VanillaEffects::HEALTH_BOOST());
        $this->sendActivation($victim, "§bPoisoned §aActivated!");
        $this->sendActivation($attacker, "§cStruck by §bPoisoned §cEnchant! Health Effects removed");
        $effect = new EffectInstance(VanillaEffects::POISON(), $enchantmentLevel * 2 * 20, $this->getLevel($enchantmentLevel), false);
        $attacker->getEffects()->add($effect);
    }

}

File: SkyblockCore/src/main/java/dev/skyblock/enchants/armor/Poisoned.java
package dev.skyblock.enchants.armor;

import dev.skyblock.enchants.BaseArmorEnchant;
import org.bukkit.entity.Player;
import org.bukkit.event.entity.EntityDamageByEntityEvent;
import org.bukkit.potion.PotionEffect;
import org.bukkit.potion.PotionEffectType;
import java.util.concurrent.ThreadLocalRandom;

public class Poisoned extends BaseArmorEnchant {
    public static final int ID = 125;

    public Poisoned() {
        super(ID, "Poisoned", 15);
    }

    @Override
    public boolean isApplicableTo(Player player, int level) {
        return ThreadLocalRandom.current().nextInt(1, 26) == 1;
    }

    @Override
    public void onActivation(Player victim, EntityDamageByEntityEvent ev, int level) {
        if (ev.getDamager() instanceof Player attacker) {
            attacker.removePotionEffect(PotionEffectType.REGENERATION);
            attacker.removePotionEffect(PotionEffectType.HEALTH_BOOST);
            victim.sendMessage("§bPoisoned §aActivated!");
            attacker.sendMessage("§cStruck by §bPoisoned §cEnchant! Health Effects removed");
            int amplifier = Math.min(5, (int) Math.ceil(level / 3.0));
            attacker.addPotionEffect(new PotionEffect(PotionEffectType.POISON, level * 2 * 20, amplifier));
        }
    }
}
